src/components/timeline/TimelineContextMenu.svelte

```diff
@@ -0,0 +1,120 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import type { ScaleTime } from 'd3-scale';
+  import { createEventDispatcher } from 'svelte';
+  import { view } from '../../stores/views';
+  import type { ActivityDirective, ActivityDirectivesMap } from '../../types/activity';
+  import type { Span, SpansMap, SpanUtilityMaps } from '../../types/simulation';
+  import type { MouseOver, VerticalGuide } from '../../types/timeline';
+  import { getAllSpansForActivityDirective, getSpanRootParent } from '../../utilities/activities';
+  import { getDoyTime, getIntervalInMs, getUnixEpochTimeFromInterval } from '../../utilities/time';
+  import { createVerticalGuide } from '../../utilities/timeline';
+  import ContextMenu from '../context-menu/ContextMenu.svelte';
+  import ContextMenuItem from '../context-menu/ContextMenuItem.svelte';
+  import ContextSubMenuItem from '../context-menu/ContextSubMenuItem.svelte';
+
+  export let activityDirectivesMap: ActivityDirectivesMap;
+  export let spansMap: SpansMap;
+  export let spanUtilityMaps: SpanUtilityMaps;
+  export let planStartTimeYmd: string;
+  export let contextMenu: MouseOver | undefined;
+  export let verticalGuides: VerticalGuide[];
+  export let xScaleView: ScaleTime<number, number> | null = null;
+
+  const dispatch = createEventDispatcher();
+  let contextMenuComponent: ContextMenu;
+  let activityDirective: ActivityDirective;
+  let activityDirectiveSpans: Span[] = [];
+  let span: Span;
+
+  // TODO imports here could be better, should we handle the vertical guide creation in Timeline?
```

Comment:
Yeah that or bubbling vertical guide creation up to a higher component since we do that in three separate places – TimelineCursors, TimelineEditorPanel, and now TimelineContextMenu. But could be a future refactor if we do anything about it?